pkg/reconciler/instances/istio/actions/actions.go

```diff
@@ -0,0 +1,146 @@
+package actions
+
+import (
+	"encoding/json"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/file"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/instances/istio/istioctl"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/kubernetes"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/types"
+	"os"
+	"strings"
+)
+
+const (
+	istioctlBinaryPathEnvKey = "ISTIOCTL_PATH"
+)
+
+type webhookPatchJSON struct {
+	Op    string                `json:"op"`
+	Path  string                `json:"path"`
+	Value webhookPatchJSONValue `json:"value"`
+}
+
+type webhookPatchJSONValue struct {
+	Key      string   `json:"key"`
+	Operator string   `json:"operator"`
+	Values   []string `json:"values"`
+}
+
+//go:generate mockery -name=IstioPerformer
+// IstioPerformer performs actions on Istio component on the cluster.
+type IstioPerformer interface {
+
+	// Install Istio on the cluster.
+	Install() error
+
+	// PatchMutatingWebhook configuration.
+	PatchMutatingWebhook() error
+}
+
+// DefaultIstioPerformer provides a default implementation of IstioPerformer.
+type DefaultIstioPerformer struct {
+	commander     istioctl.Commander
+	istioctlPath  string
+	kubeConfig    string
+	istioOperator string
+	kubeClient    kubernetes.Client
+	logger        *zap.SugaredLogger
+}
+
+// NewDefaultIstioPerformer creates a new instance of the DefaultIstioPerformer.
+func NewDefaultIstioPerformer(kubeConfig, manifest string, kubeClient kubernetes.Client, logger *zap.SugaredLogger, cmder istioctl.Commander) (*DefaultIstioPerformer, error) {
+	istioctlPath := resolveIstioctlPath()
+	if istioctlPath == "" {
+		return nil, errors.New("Istioctl binary could not be found under ISTIOCTL_PATH env variable")
+	}
+
+	istioOperator := extractIstioOperatorContextFrom(manifest)
+	if istioOperator == "" {
+		return nil, errors.New("Istio Operator definition could not be found in manifest")
```

Comment:
Same here, better to return the err directly from the `extractIstioOperatorContextFrom` to have it properly encapsulated